bitnami/redis/templates/master/statefulset.yaml

```diff
@@ -339,31 +360,31 @@ spec:
           emptyDir: {}
         - name: tmp
           emptyDir: {}
-        {{- if .Values.master.extraVolumes }}
-        {{- include "common.tplvalues.render" ( dict "value" .Values.master.extraVolumes "context" $ ) | nindent 8 }}
-        {{- end }}
         {{- if .Values.tls.enabled }}
         - name: redis-certificates
           secret:
             secretName: {{ required "A secret containing the certificates for the TLS traffic is required when TLS in enabled" .Values.tls.certificatesSecret }}
             defaultMode: 256
         {{- end }}
-  {{- if and .Values.master.persistence.enabled (not .Values.persistence.existingClaim) (not .Values.master.persistence.volumes) }}
+        {{- if .Values.master.extraVolumes }}
+        {{- include "common.tplvalues.render" ( dict "value" .Values.master.extraVolumes "context" $ ) | nindent 8 }}
+        {{- end }}
+  {{- if not .Values.master.persistence.enabled }}
+        - name: redis-data
+          emptyDir: {}
+  {{- else if .Values.master.persistence.existingClaim }}
+        - name: redis-data
+          persistentVolumeClaim:
+            claimName: {{ printf "%s" (tpl .Values.master.persistence.existingClaim .) }}
+  {{- else }}
   volumeClaimTemplates:
     - metadata:
         name: redis-data
-        labels:
-          app: {{ template "redis.name" . }}
-          release: {{ .Release.Name }}
-          heritage: {{ .Release.Service }}
-          component: master
-        {{- if .Values.master.statefulset.volumeClaimTemplates.labels }}
-        {{- toYaml .Values.master.statefulset.volumeClaimTemplates.labels | nindent 10 }}
+        labels: {{- include "common.labels.standard" . | nindent 10 }}
```

Comment:
common.labels.standard may break upgrades, because it includes the chart version and would affect an immutable field

Reply:
Good catch! Changing it to "common.labels.matchLabels"